Marlin/src/HAL/AVR/fast_pwm.cpp

```diff
@@ -257,15 +257,16 @@ void set_pwm_duty(const pin_t pin, const uint16_t v, const uint16_t v_size/*=255
       Timer timer = get_pwm_timer(pin);
       if (timer.n == 0) return; // Don't proceed if protected timer or not recognized
       // Set compare output mode to CLEAR -> SET or SET -> CLEAR (if inverted)
-      _SET_COMnQ(timer.TCCRnQ, (timer.q
+      _SET_COMnQ(timer.TCCRnQ,
+        ( timer.q
           #ifdef TCCR2
             + (timer.q == 2) // COM20 is on bit 4 of TCCR2, thus requires q + 1 in the macro
           #endif
         ), COM_CLEAR_SET + invert
       );
 
-      uint16_t top = (timer.n == 2) ? TERN(USE_OCR2A_AS_TOP, *timer.OCRnQ[0], 255) : *timer.ICRn;
-      _SET_OCRnQ(timer.OCRnQ, timer.q, (v * top + v_size / 2) / v_size); // Scale 8/16-bit v to top value
+      const uint16_t top = timer.n == 2 ? TERN(USE_OCR2A_AS_TOP, *timer.OCRnQ[0], 255) : *timer.ICRn;
+      _SET_OCRnQ(timer.OCRnQ, timer.q, v * (top / v_size)); // Scale 8/16-bit v to top value
```

Comment:
Integer division performed ahead of multiplication increases the chance of lost precision. If precision is desired, multiplication should come ahead of division, extending to 32 bits if values over 256 are expected.

Reply:
Noted, thanks.